### DISCUSSION TITLE
text_box with color

### DISCUSSION THREAD

**Discussion by obfuscoder** [prawnpdf/prawn#1139] [General]
I am using text_box as it allows me to define a box which the text resides in and will never escape from. I also want to change the color of the font. However, the color option is not supported and I have no idea how I would be able to change the color while using text_box. The documentation does not provide a solution to this problem.

I would recommend either to enhance the documentation to explain, how coloring of the text can be achieved or add support for that option to that method.

**Comment by mojavelinux** [comment]
It's necessary to use formatted_text_box and put the color on each fragment.

```rb
formatted_text_box([{ text: 'foobar', color: 'FF0000' }])
```

You could introduce a convenience function that wraps text_box:

```rb
def text_box_with_color string, opts = {}
  fragments = [{ text: string, color: opts.delete(:color) }]
  formatted_text_box fragments, opts
end

text_box_with_color 'foobar', color: 'FF0000'
```

**Comment by JohnSmall** [comment]
But how you get a background color?

**Reply by mojavelinux** [reply]
It's necessary to use a callback that draws a box using the boundaries of the fragment. You can study Asciidoctor PDF to learn how you might approach it. See https://github.com/asciidoctor/asciidoctor-pdf/blob/master/lib/asciidoctor/pdf/formatted_text/text_background_and_border_renderer.rb
